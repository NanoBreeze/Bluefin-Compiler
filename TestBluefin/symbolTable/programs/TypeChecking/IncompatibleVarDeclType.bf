/*
Verify error messages when the rhs of a variable declaration doesn't match with specified type
*/

struct A {
	float f() {
		return 4;
	}
};

int main() {

	A first;
	A second;

	int a = 5+6*false > true; 

	int b = 6.2;  // narrowing is not allowed

	A c = "hi"; // invalid

	A d = first; // valid, actually

	bool e = true && false || 1; // rhs is not a bool type since '1' is an int

	float f = first.f(); //valid 

	float g = a = b = 34; // ok

	int h = g = 5;

	return 0;
}