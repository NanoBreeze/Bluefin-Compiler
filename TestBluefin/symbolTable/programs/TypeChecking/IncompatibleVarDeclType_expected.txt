Operator '=' cannot be applied to 'int' and 'float'
Operator '=' cannot be applied to 'int' and 'float'
Operator '*' cannot be applied to 'int' and 'bool'
Operator '=' cannot be applied to 'int' and 'float'
Operator '=' cannot be applied to 'A' and 'string'
Operator '||' cannot be applied to 'bool' and 'int'
Operator '=' cannot be applied to 'int' and 'float'
